fili-core/src/main/java/com/yahoo/bard/webservice/web/responseprocessors/PartialDataV2ResponseProcessor.java

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.responseprocessors;
+
+import com.yahoo.bard.webservice.druid.client.FailureCallback;
+import com.yahoo.bard.webservice.druid.client.HttpErrorCallback;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Response processor for partial data V2.
+ * <p>
+ * If Fili expects data that is not returned from Druid, it returns an error.
+ */
+public class PartialDataV2ResponseProcessor implements FullResponseProcessor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PartialDataV2ResponseProcessor.class);
+
+    private final ResponseProcessor next;
+
+    /**
+     * Constructor.
+     *
+     * @param next  Next ResponseProcessor in the chain
+     */
+    public PartialDataV2ResponseProcessor(ResponseProcessor next) {
+        this.next = next;
+    }
+
+    @Override
+    public ResponseContext getResponseContext() {
+        return next.getResponseContext();
+    }
+
+    @Override
+    public FailureCallback getFailureCallback(DruidAggregationQuery<?> druidQuery) {
+        return next.getFailureCallback(druidQuery);
+    }
+
+    @Override
+    public HttpErrorCallback getErrorCallback(DruidAggregationQuery<?> druidQuery) {
+        return next.getErrorCallback(druidQuery);
+    }
+
+    @Override
+    public void processResponse(JsonNode json, DruidAggregationQuery<?> query, LoggingContext metadata) {
+        validateJsonResponse(json);
```

Comment:
This doesn't handle the validator throwing an exception. If this response processor decides to fail the request, _it_ needs to handle doing that failure, rather than letting someone else handle it. Likely it should call either the error or failure callback (depending on which makes more sense).